src/pages/ValidateLoginPage/index.website.tsx

```diff
@@ -18,13 +18,17 @@ function ValidateLoginPage({
         params: {accountID, validateCode, exitTo},
     },
     session,
+    autoAuthState: autoAuthStateProp,
 }: ValidateLoginPageProps<ValidateLoginPageOnyxProps>) {
     const login = credentials?.login;
-    const autoAuthState = session?.autoAuthState ?? CONST.AUTO_AUTH_STATE.NOT_STARTED;
     const isSignedIn = !!session?.authToken && session?.authTokenType !== CONST.AUTH_TOKEN_TYPES.ANONYMOUS;
+    // We don't want the previous autoAuthState affects the rendering of the current magic link page, so the autoAuthState prop sets initWithStoredValues as false,
+    // except if the user is signed in because the page will be remounted when successfully signed in as explained in Session.initAutoAuthState.
+    const autoAuthState = isSignedIn ? session?.autoAuthState : autoAuthStateProp;
```

Comment:
Having autoAuthState as undefined initially at first render works fine except when the user is successfully logged in from the magic link.

Reply:
The rest of the changes is just renaming.